PolyEngine/CMakeLists.txt

```diff
@@ -0,0 +1,100 @@
+cmake_minimum_required(VERSION 3.7)
+project(PolyEngine VERSION 0.0.1 LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+if($ENV{TRAVIS})
+	if($ENV{TRAVIS} STREQUAL "true")
+		set(IS_TRAVIS_BUILD TRUE)
+	endif($ENV{TRAVIS} STREQUAL "true")
+endif($ENV{TRAVIS})
+
+#Options and whatnot
+option(SIMD "Enable SSE3 extensions" ON)
+set(SANITIZER none CACHE STRING "C++ Sanitizer. Possible values: none, address, undefined, memory, thread") #TODO(vuko)
+set(MEMSAN_LIBCXX "" CACHE STRING "Instrumented C++ standard library location for MemorySanitizer") #TODO(vuko)
+option(CLANG_TIDY "Enable clang-tidy static analyzer" OFF) #TODO(vuko): clang-tidy -checks='cppcoreguidelines-*,clang-analyzer-*,-clang-analyzer-alpha*,-clang-analyzer-llvm.Conventions,-clang-analyzer-optin*,-clang-analyzer-osx,misc-*,modernize-*,performance-*,readability-*,google-explicit-constructor,google-readability-casting,google-runtime-int,google-runtime-memset,google-runtime-operator,llvm-namespace-comment' (maybe minus cppcoreguidelines-pro-type-union-access?)
+set(CLANG_TIDY_EXE "" CACHE STRING "clang-tidy executable location") #TODO(vuko)
+option(BUILD_TESTS "Determines whether to build tests" ON)
+option(GENERATE_COVERAGE "Enable line coverage info generated from unit test runs" OFF)
+option(GENERATE_DOXYGEN "Generate API documentation if Doxygen is available" ON)
+
+if(GENERATE_COVERAGE)
+	if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU") #TODO(vuko): clang and MSVC
+		# this is an undocumented option for changing the output extension from *.cpp.ext
+		# to *.ext; As there is no reasonable way to obtain the list of object files
+		# this seems to be the best option to generate the naming gcov expects.
+		set(CMAKE_CXX_OUTPUT_EXTENSION_REPLACE 1)
+		set(GCOV_COMMAND gcov CACHE STRING "Command to invoke to run gcov.")
+	endif()
+endif()
+
+#target_compile_features(<target name> PUBLIC cxx_static_assert) #TODO(vuko): will this be useful?
+
+#Build switches
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang") #TODO(vuko): cleanup, build types (debug, etc)
+	add_compile_options(-pipe)
+	add_compile_options(-Wall)
+	add_compile_options(-Wextra)
+	#add_compile_options(-stdlib=libc++)
+	#add_compile_options(-lc++abi)
+	if(SIMD)
+		add_compile_options(-msse3)
+	endif(SIMD)
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
Yes, as stated I want to add the support in my next PR (get the changes in first, then iterate on that to improve)